File: DSA/permutation.py
def are_perm(string1, string2):
    '''
    Compares if two strings are permutations of the other.
    Takes O(n*log(n)) because of sort.
    Note: it's case and whitespace sensitive.
    '''
    if len(string1) != len(string2):
        return False

    return sorted(string1) == sorted(string2)


def are_perm2(string1, string2):
    '''
    compares if one string is the perm of the other.
    Should take O(n). 
    Note: it's case and whitespace sensitive
    '''
    str_len = len(string1)
    if str_len != len(string2):
        return False

    str1_sum, str2_sum = 0, 0
    for i in range(str_len):
        str1_sum += ord(string1[i])
        str2_sum += ord(string2[i])

    return str1_sum == str2_sum


if __name__ == "__main__":
    str1 = "God" #'esmond'
    str2 = "dog" #'domens'
    print(str1, str2, are_perm2(str1, str2))

File: DSA/permutation.c
#include <stdio.h>
#include <string.h>

/**
 * are_perm - compares is two strings are permutations of 
 * each other
 * @s: first string
 * @t: second string
 * Return - True (1) if `are_perm` else False (0)
 */
int are_perm(char *s, char *t)
{
	int str1_sum, str2_sum, len_str = strlen(s);

	if (len_str != strlen(t))
		return (0);	
	
	str1_sum = 0;
	str2_sum = 0;
	while (*s && *t)
	{
		str1_sum += *s;
		str2_sum += *t;
		s++;
		t++;
	}
	
	if (str1_sum == str2_sum)
		return (1);
	return (0);
}


int main(int args, char* argv[])
{
	if (args <= 1)
	{
		printf("Format %s {word1} {word2}\n", argv[0]);
		return (-1);
	}
	char *word1 = argv[1];
	char *word2 = argv[2];

	printf("%s, %s, %d\n", word1, word2, are_perm(word1, word2));

	return (0);
}
